src/actions/check-solution/index.ts

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+import * as core from '@actions/core';
+import { ActionLogger, AuthHandler, AuthKind, getWorkingDirectory, PacRunner } from '../../lib';
+import path = require('path');
+import fs = require('fs-extra');
+
+core.startGroup('check-solution:');
+const workingDir = getWorkingDirectory('working-directory', false);
+const solutionPathCandidate = core.getInput('path', { required: true });
+const solutionPath = path.isAbsolute(solutionPathCandidate) ? solutionPathCandidate : path.resolve(workingDir, solutionPathCandidate);
+fs.ensureFileSync(solutionPath);
+core.info(`solution path: ${solutionPath}`);
+
+const outputDirectoryCandidate = core.getInput('output-directory', { required: true });
+const outputDirectory = path.isAbsolute(outputDirectoryCandidate) ? outputDirectoryCandidate : path.resolve(workingDir, outputDirectoryCandidate);
+fs.ensureDirSync(outputDirectory);
+core.info(`output directory: ${solutionPath}`);
+
+const geo = core.getInput('geo', { required: false });
+const ruleLevelOverride = core.getInput('rule-level-override', { required: false });
+
+const logger = new ActionLogger();
+
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+(async () => {
+    const pac = new PacRunner(workingDir, logger);
+    await new AuthHandler(pac).authenticate(AuthKind.CDS);
+
+    const checkArgs = ['solution', 'check', '--path', solutionPath, '--outputDirectory', outputDirectory];
+    if (geo) {
+        checkArgs.push('--geo', geo);
+    }
+    if (ruleLevelOverride) {
+        checkArgs.push('--ruleLevelOverride', ruleLevelOverride);
+    }
+    await pac.run(checkArgs);
+    core.info(`checked solution results in folder: ${outputDirectory}`);
```

Comment:
thinking out loud, not sure what is best here:
since the .sarif and summary files are the main results, should we instead not ask the workflow author for an output directory, but instead for an artifact name instead? That way, this action will directly upload the checker result files into an artifact folder or zip, without even requiring the workflow author to add a separate upload-artifact action

Reply:
Removed the `output-directory ` argument, and uploaded the artifacts.